compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -373,8 +381,17 @@ execCompile inputFile outputFile opts =
                 files <- nubSort . concatMap transitiveModuleDeps <$> use GetDependencies inputFile
                 mbIfaces <- writeIfacesAndHie (toNormalizedFilePath $ fromMaybe ifaceDir $ optIfaceDir opts') files
                 void $ liftIO $ mbErr "ERROR: Compilation failed." mbIfaces
+
             mbDalf <- getDalf inputFile
             dalf <- liftIO $ mbErr "ERROR: Compilation failed." mbDalf
+
+            -- Support for '-ddump-parsed', '-ddump-parsed-ast', '-dsource-stats'.
+            dflags <- hsc_dflags . hscEnv <$> use_ GhcSession inputFile
+            parsed <- pm_parsed_source <$> use_ GetParsedModule inputFile
+            liftIO $ ErrUtils.dumpIfSet_dyn dflags Opt_D_dump_parsed "Parser" $ GHC.ppr parsed
```

Comment:
We could wrap this into one `liftIO` black, like
```
liftIO $ do
  ErrUtils.dumpIfSet_dyn ...
  ErrUtils.dumpIfSet_dyn ...
  ...
```
if you like that more.

Reply:
Ooh. Definitely. Will fix.